include/libopencm3/stm32/f4/adc.h

```diff
@@ -1,200 +1,6 @@
-/** Timer 3 Compare Output 1 */
-#define ADC_CR2_EXTSEL_TIM3_CC1		(0x7 << 24)
-/** Timer 3 TRGO Event */
-#define ADC_CR2_EXTSEL_TIM3_TRGO	(0x8 << 24)
-/** Timer 4 Compare Output 4 */
-#define ADC_CR2_EXTSEL_TIM4_CC4		(0x9 << 24)
-/** Timer 5 Compare Output 1 */
-#define ADC_CR2_EXTSEL_TIM5_CC1		(0xA << 24)
-/** Timer 5 Compare Output 2 */
-#define ADC_CR2_EXTSEL_TIM5_CC2		(0xB << 24)
-/** Timer 5 Compare Output 3 */
-#define ADC_CR2_EXTSEL_TIM5_CC3		(0xC << 24)
-/** Timer 8 Compare Output 1 */
-#define ADC_CR2_EXTSEL_TIM8_CC1		(0xD << 24)
-/** Timer 8 TRGO Event */
-#define ADC_CR2_EXTSEL_TIM8_TRGO	(0xE << 24)
-/** EXTI Line 11 Event */
-#define ADC_CR2_EXTSEL_EXTI_LINE_11	(0xF << 24)
-/**@}*/
-
-/* JEXTSEL[3:0]: External event selection for injected group. */
-/****************************************************************************/
-/** @defgroup adc_trigger_injected ADC Trigger Identifier for Injected group
-@ingroup adc_defines
-
-@{*/
-#define ADC_CR2_JEXTSEL_TIM1_CC4	(0x0 << 16)
-#define ADC_CR2_JEXTSEL_TIM1_TRGO	(0x1 << 16)
-#define ADC_CR2_JEXTSEL_TIM2_CC1	(0x2 << 16)
-#define ADC_CR2_JEXTSEL_TIM2_TRGO	(0x3 << 16)
-#define ADC_CR2_JEXTSEL_TIM3_CC2	(0x4 << 16)
-#define ADC_CR2_JEXTSEL_TIM3_CC4	(0x5 << 16)
-#define ADC_CR2_JEXTSEL_TIM4_CC1	(0x6 << 16)
-#define ADC_CR2_JEXTSEL_TIM4_CC2	(0x7 << 16)
-#define ADC_CR2_JEXTSEL_TIM4_CC3	(0x8 << 16)
-#define ADC_CR2_JEXTSEL_TIM4_TRGO	(0x9 << 16)
-#define ADC_CR2_JEXTSEL_TIM5_CC4	(0xA << 16)
-#define ADC_CR2_JEXTSEL_TIM5_TRGO	(0xB << 16)
-#define ADC_CR2_JEXTSEL_TIM8_CC2	(0xC << 16)
-#define ADC_CR2_JEXTSEL_TIM8_CC3	(0xD << 16)
-#define ADC_CR2_JEXTSEL_TIM8_CC4	(0xE << 16)
-#define ADC_CR2_JEXTSEL_EXTI_LINE_15	(0xF << 16)
-/**@}*/
-
-/* ADC_SMPRG ADC Sample Time Selection for Channels */
-/** @defgroup adc_sample_rg ADC Sample Time Selection for All Channels
-@ingroup adc_defines
-
-@{*/
-#define ADC_SMPR_SMP_3CYC		0x0
-#define ADC_SMPR_SMP_15CYC		0x1
-#define ADC_SMPR_SMP_28CYC		0x2
-#define ADC_SMPR_SMP_56CYC		0x3
-#define ADC_SMPR_SMP_84CYC		0x4
-#define ADC_SMPR_SMP_112CYC		0x5
-#define ADC_SMPR_SMP_144CYC		0x6
-#define ADC_SMPR_SMP_480CYC		0x7
-/**@}*/
-
-/* --- ADC_SQR1 values ----------------------------------------------------- */
-#define ADC_SQR1_L_MSK			(0xf << ADC_SQR1_L_LSB)
-
-#define ADC_SQR_MAX_CHANNELS_REGULAR	16
-
-/* ADCPRE: ADC prescaler. */
-/****************************************************************************/
-/** @defgroup adc_ccr_adcpre ADC Prescale
-@ingroup adc_defines
-
-@{*/
-#define ADC_CCR_ADCPRE_BY2		(0x0 << 16)
-#define ADC_CCR_ADCPRE_BY4		(0x1 << 16)
-#define ADC_CCR_ADCPRE_BY6		(0x2 << 16)
-#define ADC_CCR_ADCPRE_BY8		(0x3 << 16)
-/**@}*/
-#define ADC_CCR_ADCPRE_MASK		(0x3 << 16)
-#define ADC_CCR_ADCPRE_SHIFT		16
-
-
-BEGIN_DECLS
-
-void adc_set_multi_mode(uint32_t mode);
-void adc_enable_vbat_sensor(void);
-void adc_disable_vbat_sensor(void);
-
-END_DECLS
+#include <libopencm3/stm32/common/dma_common_f24.h>
```

Comment:
sure about this?

Reply:
d'oh! I only built my project with F2 support, so I missed this erroneous copy-paste.

Reply:
That's one of the reasons I was against supporting people building only part of the project :)